Examples/TodoApp/TodoAppTests/src/tasks/effecthandlers/TasksListEffectHandlerTests.swift

```diff
@@ -0,0 +1,28 @@
+import XCTest
+@testable import TodoApp
+import MobiusCore
+
+final class TasksListEffectHandlerTests: XCTestCase {
+
+    private var subject: TasksListEffectHandler!
+
+    override func setUp() {
+        super.setUp()
+        subject = TasksListEffectHandler()
+    }
+
+    override func tearDown() {
+        subject = nil
+        super.tearDown()
+    }
+
+    // MARK: - Tests
+
+    func testConnect() {
```

Comment:
this is left empty on purpose, because with the current signature we can't test and effect handler AFAIK
looking for feedback

Reply:
The effect handler is a `Connectable`, so in your test you should call `.connect` and pass in a closure which receives its output events.
If you want to test that side-effects outside of the loop are performed, inject fakes and assert that they end up in the expected state.

Reply:
The problem is that in our current implementation we are not using the only input parameter of the only public function we have: the `eventConsumer`. We probably didn't understand how it's supposed to be used here.

If I'm not wrong, with our current implementation, this can't be tested, since you can't control which events are being sent.

```    func connect(_ eventConsumer: @escaping Consumer<TasksList.Event>) -> Connection<TasksList.Effect> {
        return Connection<TasksList.Effect>(
            acceptClosure: routeEffect,
            disposeClosure: { /* Nothing to dispose */ }
        )
    }
```

Reply:
But you are injecting a `ViewModel`, so create a fake one of those and verify that the expected things happen when you send effects.

Side note - the `eventConsumer` is meant to be used to send things back to the loop if needed. So, for instance, your effect handler could fetch some data, then send an event back to the loop when its done using the `eventConsumer`